common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -801,9 +807,13 @@ def new_lcp(self, state, text=None):
         if text is None:
             text = self.data
 
+        user_service = self.runtime.service(self, 'user')
+        anonymous_student_id = user_service.get_current_user().opt_attrs.get(ATTR_KEY_ANONYMOUS_USER_ID)
```

Comment:
@symbolist CC @ormsbee  Here's the line that caused the issue, in [DjangoXBlockUserService.__init__](https://github.com/edx/edx-platform/pull/28440/files#diff-4e10e4b8dffa69869f2098d333e9dc71c7e56b82cd08529bec0cc337aab9b724R43)

That same `django_user` object gets passed through when creating the ProblemBlock's user service, and its parent Vertical and Sequence blocks. I unwisely stored the `anonymous_user_id` attribute on that shared `django_user` object (mimicking the [`user_is_staff` attribute storage in the line above](https://github.com/edx/edx-platform/pull/28440/files#diff-4e10e4b8dffa69869f2098d333e9dc71c7e56b82cd08529bec0cc337aab9b724R42)), and since the `anonymous_user_id` value is different for Problem/HTML blocks vs Vertical/Sequence blocks (cf [`requires_per_student_anonymous_id`](https://github.com/edx/edx-platform/pull/28440/files#diff-3b0c1e091e61826c4448bc5924e2911dedfd829754b2c6087e7067dfbdeed895R547)), it was being altered during rendering.

https://github.com/edx/edx-platform/pull/29190 will address this issue and add a test to verify the fix.